libraries/chain/db_getter.cpp

```diff
@@ -251,7 +252,7 @@ bool database::is_son_dereg_valid( const son_id_type& son_id )
    const auto& son_idx = get_index_type<son_index>().indices().get< by_id >();
    auto son = son_idx.find( son_id );
    FC_ASSERT( son != son_idx.end() );
-   bool ret = ( son->status == son_status::in_maintenance &&
+   bool ret = ( ((son->status == son_status::in_maintenance) || (son->status == son_status::request_maintenance)) &&
```

Comment:
Same here, dereg is valid for a SON which is in_maintenance for at least 12 hours.

Reply:
It is for the case when maintenance block is going after 12 hours after the start of the request_maintenance state (some error in network) and SON wasn't moved to in_maintenance state,  but 12 hours are passed.

Reply:
Actually it is expected that the SON continues to send heartbeats in request_maintenance state. So it is expected that the SON continues to work in full capacity till the next chain maintenance.